geopandas/tests/test_geodataframe.py

```diff
@@ -372,6 +371,12 @@ def test_get_geometry_geometry_inactive(self):
         with pytest.raises(AttributeError, match=msg_geo_col_missing):
             df.geometry
 
+    def test_geoseries_override_existing_crs_warning(self):
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Use `with pytest.warns` instead. Check for other cases of that in tests to see how we use it.